src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,53 @@
+import cn from 'classnames';
+import React from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  onDeleteTodo: (todoId: number) => Promise<any>,
+  isDeleting: boolean;
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  onDeleteTodo,
+  isDeleting,
+}) => {
+  const isLoading = todo.id === 0 || isDeleting;
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', { completed: todo.completed })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+        />
+      </label>
+
+      <span data-cy="TodoTitle" className="todo__title">
+        {todo.title}
+      </span>
+      <button
+        type="button"
+        className="todo__remove"
+        data-cy="TodoDeleteButton"
+        onClick={() => onDeleteTodo(todo.id)}
```

Comment:
The function name should describe what it does. So `deleteTodo` is a better name than `onDelete...`
Fix it please in all places
```suggestion
        onClick={() => deleteTodo(todo.id)}
```